### ISSUE TITLE
Agents very slow

### ISSUE THREAD

**Issue by alminveh** [actions/runner-images#8380]
### Description

Since September 21 agents are very slow. Even simple tasks that usually take 1-2 seconds now take 25-30 seconds. A comparison of some tasks from two runs can be seen below:
![image](https://github.com/actions/runner-images/assets/133762074/649ded77-ef00-450a-9c3d-fa5c3aa24c8a)
On the left side we have a run before September 21 (simple tasks taking 1-2 seconds to complete), on the right side we have the same tasks running on September 25 (same tasks taking 25-30 seconds to complete.

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [X] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

Issue started from version 20230918.1.0

### Is it regression?

20230912.1.0

### Expected behavior

Tasks are fast to execute

### Actual behavior

Task execution is very slow

### Repro steps

-

**Comment by ilia-shipitsin** [comment]
can you please provide repro steps ? what is "fetch step-template commit id" ?

**Comment by alminveh** [comment]
@ilia-shipitsin 
Taking "Create temp-dir folder and set the current year" as example, it went from 1s to 28s, and the task is this:
`    - powershell: |
         md $(_tempFolder)\versionFile;
         $year=(Get-Date).ToString('yyyy')
         Write-Host "##vso[task.setvariable variable=currentYear;]$year"
       displayName: 'Create temp-dir folder and set the current year' `

@mikhailkoliada 
Does not seem like a networking issue. We have a run from last Wednesday (Sept 20), six agents executing the same jobs in parallel, five of them are agents with image version 20230912.1.0 and one with 20230918.1.0 -> five agents with 20230912.1.0 have similar execution time for all tasks, one with 20230918.1.0 takes way more time for almost all tasks.

**Comment by ilia-shipitsin** [comment]
@alminveh , I added step like you described to my pipeline

GUI shows ~35s

![image](https://github.com/actions/runner-images/assets/125650415/5206c011-1b41-46f8-9eb0-dfdb9f37631d)

**Comment by alminveh** [comment]
@ilia-shipitsin is that with 20230918.1.0 image? Because with all the runs I have with 20230912.1.0 it took either one or two seconds.

**Comment by ilia-shipitsin** [comment]
yes, it is mentioned image. I do not believe image is related, but we can reproduce and investigate.

![image](https://github.com/actions/runner-images/assets/125650415/f79d7705-2d98-4a3f-ac58-afdf44602343)

**Comment by ilia-shipitsin** [comment]
I collected logs, I see that "windows update" takes 15 sec

![image](https://github.com/actions/runner-images/assets/125650415/0798d9c5-ee9e-4942-aad1-6c370ca67d2d)


it did not suppose to happen, because we disable windows update:

https://github.com/actions/runner-images/blob/main/images/win/scripts/Installers/Finalize-VM.ps1#L94

interesting thing is "windows update medic service" which seems to enable windows update back

![image](https://github.com/actions/runner-images/assets/125650415/196fa945-246a-4638-b62f-256298169c08)

**Comment by iglendd** [comment]
I am second that. The same timeframe (Sep 20) and the same runner (20230918) make 8 runs of Powershell scripts which change the ACL of a file, previously (20230910) took 10-20 seconds and now are taking 3 min or more. Do not know how much more yet since we have a 3 min timeout window.

**Comment by ilia-shipitsin** [comment]
@iglendd , sorry, I cannot say either your issue is similar or not. Feel free to open separate issue with full repro steps. We'll investigate and mention if issues are related.

**Comment by ilia-shipitsin** [comment]
I planned to disable WaasMedicSvc like we usually disable services, seems it will require more attention ((

![image](https://github.com/actions/runner-images/assets/125650415/787c4e10-0eb9-4519-be84-d03e0c9d610e)

**Comment by QiuyongWang** [comment]
Hi team, is there any ETA plan to fix this issue? thanks.

**Comment by ilia-shipitsin** [comment]
let me summarize findings

1) WaasMedicSvc is not expected here, but it is not the root cause, we'll address it later.


what was found

something has changed in powershell.exe (which is Powershell 5.1) behaviour. It "analyzes" all possible modules during startup and builds a cache. Somehow it started to take ~20 sec (we did not change modules a lot, maybe few minor changes if any at all)

https://learn.microsoft.com/en-us/powershell/scripting/windows-powershell/wmf/whats-new/release-notes?view=powershell-7.3#module-analysis-cache - this is eating ~20 sec

if you can switch to "pwsh" - it would be a workaround (in short, if you use GHA - you can, in ADO tasks are built on top of "powershell.exe", not possible to switch).

we'll try to escalate to powershell team. no idea (yet) how to fix that.

**Comment by ilia-shipitsin** [comment]
I compared current image against previous, I haven't found anything that looks suspicious

https://github.com/actions/runner-images/compare/win22/20230910.1...win22/20230918.1

**Comment by alminveh** [comment]
What we see in our runs is not that something adds 20-30 seconds (or minutes) to execution time but overall execution is just way slower. Our test run tasks (WebDriverIO tests) that use to take between 2.5 and 3 hours now consistently run for 6+ hours (and fail with agent timeout every time), shorter test runs doubled in time needed to complete (~10 min -> ~20 min, ~1 hour -> 2+ hours, etc.).

**Comment by ilia-shipitsin** [comment]
@alminveh , initially you reported that "even simple Powershell task takes 20-30 sec instead of 1 sec". I confirm that. If your tests under the hood invoke powershell many times, I suspect it is the same issue. If not, please provide repro steps (better in separate issue)

**Comment by ilia-shipitsin** [comment]
good news, `20230924.1.0` (which is currently being deployed) seem to be not affected

**Comment by alminveh** [comment]
I confirm that everything is back to normal with 20230924.1.0. Thank you @ilia-shipitsin .

**Comment by jessehouwing** [comment]
Current image version: '20230924.1.0'

Gives me random slowdowns of tasks. I've seen:

- Checkout take 2+ minutes longer on very small repo (normally 8s)
- `az cli` take 6+ minutes to reindex its command index (normally 45s)

Explicitly stopping Windows Update and StorSVC prior to checkout normalizes things:

```
steps:
  - script:
      net stop storsvc
      net stop wuauserv
  - checkout: self
```

**Comment by ilia-shipitsin** [comment]
feel free to play with disabling StorSvc for measurements.

we plan to release an image with disabled StorSvc next week: https://github.com/actions/runner-images/pull/8388

**Comment by mikhailkoliada** [comment]
Underlying problem must be solved, closing now, feel free to open separate tickets for something else
